import sys
from PyQt5 import QtCore, QtGui, QtWidgets
import math as mt;

class Ui_Calc(object):
    def setupUi(self, Calc):
        Calc.setObjectName("Calc")
        Calc.resize(332, 540)
        self.centralwidget = QtWidgets.QWidget(Calc)
        self.centralwidget.setObjectName("centralwidget")
        self.num3 = QtWidgets.QPushButton(self.centralwidget)
        self.num3.setGeometry(QtCore.QRect(250, 220, 71, 71))
        font = QtGui.QFont()
        font.setFamily("DejaVu Serif")
        font.setPointSize(28)
        font.setBold(True)
        font.setWeight(75)
        self.num3.setFont(font)
        self.num3.setStyleSheet("border-radius: 10px;\n"
"background-color: rgb(163, 159, 159);")
        self.num3.setObjectName("num3")
        self.num6 = QtWidgets.QPushButton(self.centralwidget)
        self.num6.setGeometry(QtCore.QRect(250, 300, 71, 71))
        font = QtGui.QFont()
        font.setFamily("DejaVu Serif")
        font.setPointSize(28)
        font.setBold(True)
        font.setWeight(75)
        self.num6.setFont(font)
        self.num6.setStyleSheet("border-radius: 10px;\n"
"background-color: rgb(163, 159, 159);")
        self.num6.setObjectName("num6")
        self.num9 = QtWidgets.QPushButton(self.centralwidget)
        self.num9.setGeometry(QtCore.QRect(250, 380, 71, 71))
        font = QtGui.QFont()
        font.setFamily("DejaVu Serif")
        font.setPointSize(28)
        font.setBold(True)
        font.setWeight(75)
        self.num9.setFont(font)
        self.num9.setStyleSheet("border-radius: 10px;\n"
"background-color: rgb(163, 159, 159);")
        self.num9.setObjectName("num9")
        self.num2 = QtWidgets.QPushButton(self.centralwidget)
        self.num2.setGeometry(QtCore.QRect(170, 220, 71, 71))
        font = QtGui.QFont()
        font.setFamily("DejaVu Serif")
        font.setPointSize(28)
        font.setBold(True)
        font.setWeight(75)
        self.num2.setFont(font)
        self.num2.setStyleSheet("border-radius: 10px;\n"
"background-color: rgb(163, 159, 159);")
        self.num2.setObjectName("num2")
        self.num5 = QtWidgets.QPushButton(self.centralwidget)
        self.num5.setGeometry(QtCore.QRect(170, 300, 71, 71))
        font = QtGui.QFont()
        font.setFamily("DejaVu Serif")
        font.setPointSize(28)
        font.setBold(True)
        font.setWeight(75)
        self.num5.setFont(font)
        self.num5.setStyleSheet("border-radius: 10px;\n"
"background-color: rgb(163, 159, 159);")
        self.num5.setObjectName("num5")
        self.num8 = QtWidgets.QPushButton(self.centralwidget)
        self.num8.setGeometry(QtCore.QRect(170, 380, 71, 71))
        font = QtGui.QFont()
        font.setFamily("DejaVu Serif")
        font.setPointSize(28)
        font.setBold(True)
        font.setWeight(75)
        self.num8.setFont(font)
        self.num8.setStyleSheet("border-radius: 10px;\n"
"background-color: rgb(163, 159, 159);")
        self.num8.setObjectName("num8")
        self.num0 = QtWidgets.QPushButton(self.centralwidget)
        self.num0.setGeometry(QtCore.QRect(170, 460, 71, 71))
        font = QtGui.QFont()
        font.setFamily("DejaVu Serif")
        font.setPointSize(28)
        font.setBold(True)
        font.setWeight(75)
        self.num0.setFont(font)
        self.num0.setStyleSheet("border-radius: 10px;\n"
"background-color: rgb(163, 159, 159);")
        self.num0.setObjectName("num0")
        self.num1 = QtWidgets.QPushButton(self.centralwidget)
        self.num1.setGeometry(QtCore.QRect(90, 220, 71, 71))
        font = QtGui.QFont()
        font.setFamily("DejaVu Serif")
        font.setPointSize(28)
        font.setBold(True)
        font.setWeight(75)
        self.num1.setFont(font)
        self.num1.setStyleSheet("border-radius: 10px;\n"
"background-color: rgb(163, 159, 159);")
        self.num1.setObjectName("num1")
        self.num4 = QtWidgets.QPushButton(self.centralwidget)
        self.num4.setGeometry(QtCore.QRect(90, 300, 71, 71))
        font = QtGui.QFont()
        font.setFamily("DejaVu Serif")
        font.setPointSize(28)
        font.setBold(True)
        font.setWeight(75)
        self.num4.setFont(font)
        self.num4.setStyleSheet("border-radius: 10px;\n"
"background-color: rgb(163, 159, 159);")
        self.num4.setObjectName("num4")
        self.num7 = QtWidgets.QPushButton(self.centralwidget)
        self.num7.setGeometry(QtCore.QRect(90, 380, 71, 71))
        font = QtGui.QFont()
        font.setFamily("DejaVu Serif")
        font.setPointSize(28)
        font.setBold(True)
        font.setWeight(75)
        self.num7.setFont(font)
        self.num7.setStyleSheet("border-radius: 10px;\n"
"background-color: rgb(163, 159, 159);")
        self.num7.setObjectName("num7")
        self.divide = QtWidgets.QPushButton(self.centralwidget)
        self.divide.setGeometry(QtCore.QRect(10, 460, 71, 71))
        font = QtGui.QFont()
        font.setFamily("DejaVu Serif")
        font.setPointSize(18)
        font.setBold(True)
        font.setWeight(75)
        self.divide.setFont(font)
        self.divide.setStyleSheet("border-radius: 10px;\n"
"background-color: rgb(213, 213, 213);")
        self.divide.setObjectName("divide")
        self.minus = QtWidgets.QPushButton(self.centralwidget)
        self.minus.setGeometry(QtCore.QRect(10, 220, 71, 71))
        font = QtGui.QFont()
        font.setFamily("DejaVu Serif")
        font.setPointSize(18)
        font.setBold(True)
        font.setWeight(75)
        self.minus.setFont(font)
        self.minus.setStyleSheet("border-radius: 10px;\n"
"background-color: rgb(213, 213, 213);")
        self.minus.setObjectName("minus")
        self.plus = QtWidgets.QPushButton(self.centralwidget)
        self.plus.setGeometry(QtCore.QRect(10, 300, 71, 71))
        font = QtGui.QFont()
        font.setFamily("DejaVu Serif")
        font.setPointSize(18)
        font.setBold(True)
        font.setWeight(75)
        self.plus.setFont(font)
        self.plus.setStyleSheet("border-radius: 10px;\n"
"background-color: rgb(213, 213, 213);")
        self.plus.setObjectName("plus")
        self.label = QtWidgets.QLabel(self.centralwidget)
        self.label.setGeometry(QtCore.QRect(70, 110, 200, 61))
        font = QtGui.QFont()
        font.setFamily("DejaVu Serif")
        font.setPointSize(28)
        font.setBold(False)
        font.setWeight(50)
        self.label.setFont(font)
        self.label.setObjectName("label")
        self.label_2 = QtWidgets.QLabel(self.centralwidget)
        self.label_2.setGeometry(QtCore.QRect(70, 30, 200, 61))
        font = QtGui.QFont()
        font.setFamily("DejaVu Serif")
        font.setPointSize(28)
        font.setBold(False)
        font.setWeight(50)
        self.label_2.setFont(font)
        self.label_2.setStyleSheet("color: rgb(200, 200, 200);")
        self.label_2.setObjectName("label_2")
        self.clear = QtWidgets.QPushButton(self.centralwidget)
        self.clear.setGeometry(QtCore.QRect(250, 460, 71, 71))
        font = QtGui.QFont()
        font.setFamily("DejaVu Serif")
        font.setPointSize(18)
        font.setBold(True)
        font.setWeight(75)
        self.clear.setFont(font)
        self.clear.setStyleSheet("border-radius: 10px;\n"
"background-color: rgb(213, 213, 213);")
        self.clear.setObjectName("clear")
        self.multiplication = QtWidgets.QPushButton(self.centralwidget)
        self.multiplication.setGeometry(QtCore.QRect(10, 380, 71, 71))
        font = QtGui.QFont()
        font.setFamily("DejaVu Serif")
        font.setPointSize(18)
        font.setBold(True)
        font.setWeight(75)
        self.multiplication.setFont(font)
        self.multiplication.setStyleSheet("border-radius: 10px;\n"
"background-color: rgb(213, 213, 213);")
        self.multiplication.setObjectName("multiplication")
        self.equals = QtWidgets.QPushButton(self.centralwidget)
        self.equals.setGeometry(QtCore.QRect(90, 460, 71, 71))
        font = QtGui.QFont()
        font.setFamily("DejaVu Serif")
        font.setPointSize(18)
        font.setBold(True)
        font.setWeight(75)
        self.equals.setFont(font)
        self.equals.setStyleSheet("border-radius: 10px;\n"
"background-color: rgb(213, 213, 213);")
        self.equals.setObjectName("equals")
        Calc.setCentralWidget(self.centralwidget)

        self.retranslateUi(Calc)
        QtCore.QMetaObject.connectSlotsByName(Calc)

    def retranslateUi(self, Calc):
        _translate = QtCore.QCoreApplication.translate
        Calc.setWindowTitle(_translate("Calc", "Calculator"))
        self.num3.setText(_translate("Calc", "3"))
        self.num6.setText(_translate("Calc", "6"))
        self.num9.setText(_translate("Calc", "9"))
        self.num2.setText(_translate("Calc", "2"))
        self.num5.setText(_translate("Calc", "5"))
        self.num8.setText(_translate("Calc", "8"))
        self.num0.setText(_translate("Calc", "0"))
        self.num1.setText(_translate("Calc", "1"))
        self.num4.setText(_translate("Calc", "4"))
        self.num7.setText(_translate("Calc", "7"))
        self.divide.setText(_translate("Calc", "/"))
        self.minus.setText(_translate("Calc", "-"))
        self.plus.setText(_translate("Calc", "+"))
        self.label.setText(_translate("Calc", ""))
        self.label_2.setText(_translate("Calc", "result"))
        self.clear.setText(_translate("Calc", "C"))
        self.multiplication.setText(_translate("Calc", "*"))
        self.equals.setText(_translate("Calc", "="))


# if __name__ == "__main__":
#     import sys
#     app = QtWidgets.QApplication(sys.argv)
#     Calc = QtWidgets.QMainWindow()
#     ui = Ui_Calc()
#     ui.setupUi(Calc)
#     Calc.show()
#     sys.exit(app.exec_())

class Window(QtWidgets.QMainWindow):
    def __init__(self):
        super(Window,self).__init__()
        self.ui = Ui_Calc()
        self.ui.setupUi(self)
        self.ui.num1.clicked.connect(lambda: self.NumBut(1))
        self.ui.num2.clicked.connect(lambda: self.NumBut(2))
        self.ui.num3.clicked.connect(lambda: self.NumBut(3))
        self.ui.num4.clicked.connect(lambda: self.NumBut(4))
        self.ui.num5.clicked.connect(lambda: self.NumBut(5))
        self.ui.num6.clicked.connect(lambda: self.NumBut(6))
        self.ui.num7.clicked.connect(lambda: self.NumBut(7))
        self.ui.num8.clicked.connect(lambda: self.NumBut(8))
        self.ui.num9.clicked.connect(lambda: self.NumBut(9))
        self.ui.num0.clicked.connect(lambda: self.NumBut(0))
        self.ui.clear.clicked.connect(self.Clear)
        self.ui.equals.clicked.connect(self.Equere)
        self.ui.minus.clicked.connect(lambda: self.NumBut("-"))
        self.ui.plus.clicked.connect(lambda: self.NumBut("+"))
        self.ui.divide.clicked.connect(lambda: self.NumBut("/"))
        self.ui.multiplication.clicked.connect(lambda: self.NumBut("*"))




    def Equere(self):
        minus = "-"
        plus = "+"
        split = "/"
        multiplication = "*"

        Compare = self.ui.label.text()
        if "-" in Compare:
            print("Wow")
            s = Compare.split("-")
            n = float(s[0])
            n1 = float(s[1])
            f = float(n - n1)
            self.ui.label.setText(str(mt.floor(f)))

        elif plus in Compare:
            print("Wow")
            s = Compare.split("+")
            n = float(s[0])
            n1 = float(s[1])
            f = float(n + n1)
            self.ui.label.setText(str(mt.floor(f)))

        elif split in Compare:
            print("Wow")
            s = Compare.split("/")
            n = float(s[0])
            n1 = float(s[1])
            f = float(n / n1)
            self.ui.label.setText(str(mt.floor(f)))

        elif multiplication in Compare:
            print("Wow")
            s = Compare.split("*")
            n = float(s[0])
            n1 = float(s[1])
            f = float(n * n1)
            self.ui.label.setText(str(mt.floor(f)))

        self.ui.label_2.setText(self.ui.label.text())
        self.ui.label.clear()
    def Clear(self):
        self.ui.label.clear()
    def NumBut(self,num):
        v = str(num)
        self.ui.label.setText(self.ui.label.text()+v)

app = QtWidgets.QApplication([])
applicatin = Window()
applicatin.show()
sys.exit(app.exec())
